File: backend/EpiSafeZone/src/main/java/com/example/episafezone/models/Crisis.java
package com.example.episafezone.models;
import jakarta.persistence.*;
import java.util.Date;

@Entity(name="crisis")
public class Crisis {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private Integer duration;
    @Temporal(TemporalType.DATE)
    private Date date;
    private String hour;
    private String context;
    private Boolean emergency;
    private Integer manifestation;
    private Integer patient;


    public Crisis(Integer duration, Date date, String hour, String context, Boolean emergency, Manifestation manifestation, Patient patient) {
        this.duration = duration;
        this.date = date;
        this.hour = hour;
        this.context = context;
        this.emergency = emergency;
        this.manifestation = manifestation.getId();
        this.patient = patient.getId();
    }

    public Crisis() {
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getDuration() {
        return duration;
    }

    public void setDuration(Integer duration) {
        this.duration = duration;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getHour() {
        return hour;
    }

    public void setHour(String hour) {
        this.hour = hour;
    }

    public String getContext() {
        return context;
    }

    public void setContext(String context) {
        this.context = context;
    }

    public Boolean getEmergency() {
        return emergency;
    }

    public void setEmergency(Boolean emergency) {
        this.emergency = emergency;
    }

    public Integer getManifestation() {
        return manifestation;
    }

    public void setManifestation(Integer manifestation) {
        this.manifestation = manifestation;
    }

    public Integer getPatient() {
        return patient;
    }

    public void setPatient(Integer patient) {
        this.patient = patient;
    }

    public String toString(){
        return "duración" + this.hour + "con contexto" + this.context;
    }
}

File: frontend/EpiSafeZone/app/src/main/java/com/example/episafezone/models/Crisis.kt
package com.example.episafezone.models

import java.util.Date

class Crisis (
    val id : Int,
    val duration : Int,
    val date : Date,
    val hour : String,
    val context : String,
    val emergency : Boolean,
    val manifestationName : String,
    val manifestationId : Int,
    val patient : Int) {

    constructor(duration: Int,
                date: Date,
                hour: String,
                context: String,
                emergency: Boolean,
                manifestationName: String,
                manifestationId: Int,
                patient: Int) : this(
        duration = duration,
        date = date,
        hour = hour,
        context = context,
        emergency = emergency,
        manifestationName = manifestationName,
        manifestationId = manifestationId,
        patient = patient,
        id = -1

    )
}